lib/get.js

```diff
@@ -16,30 +16,59 @@ var agent = new AgentKeepAlive({
  * @param {string} uri
  * @param {function} callback
  */
-module.exports = function get(uri, callback) {
+module.exports = function get(options, callback) {
+    var request = getClient();
+    var params = normalizeParams(options);
+    if(!params) { callback(new Error('Invalid parameters: ' + JSON.stringify(options))); }
 
     // This might load the same file multiple times, but the overhead should
     // be very little.
-    request({
-        uri: uri,
-        timeout: 5000,
-        headers: { 'accept-encoding': 'binary' },
-        encoding: 'binary',
-        // Don't use keepalive agent for https.
-        agent: uri.indexOf('https') === 0 ? null : agent
-    }, function(err, resp, data) {
+    request(params, function(err, resp, data) {
         // Use err.status of 400 as it's not an unexpected application error,
         // but likely due to a bad request. Catches ECONNREFUSED,
         // getaddrinfo ENOENT, etc.
-        if (err || !data || !resp || resp.statusCode != 200) {
-            return callback(new Error('Unable to load marker from URL.'));
+        if (err || !data || !resp || resp.statusCode !== 200) {
+            var reqErr = new Error('Unable to load marker from URL.');
+            reqErr.originalError = err;
+            return callback(reqErr);
         }
         // request 2.2.x *always* returns the response body as a string.
         // @TODO remove this once request is upgraded.
         if (!(data instanceof Buffer)) data = new Buffer(data, 'binary');
         // Restrict data length.
         if (data.length > 32768) return callback(400);
-        var headers = resp.headers;
         callback(null, data);
     });
 };
+
+function normalizeParams(options) {
+    var params = {};
+
+    if(typeof options === 'string') {
+        params.uri = options;
+    } else if(typeof options === 'object') {
+        params = xtend(params, options);
+    } else {
+        return false;
+    }
+
+    var uri = params.uri || params.url;
+    if (typeof params.agent === 'undefined') {
+        // Don't use keepalive agent for https.
+        params.agent = uri.indexOf('https') === 0 ? null : agent;
+    }
+
+    return params;
+}
+
+// allows passing in a custom request handler
+function getClient() {
+    if (module.exports.requestClient) { return module.exports.requestClient; }
+    return require('request').defaults({
```

Comment:
Why require('request') here rather then at the top?

Reply:
I do not remember - I might've been trying to mock this with testdouble or rewire at some point, or concerned about global state pollution on the request library. Doesn't seem like any of those are the case, so changed. Thanks!